### PR TITLE
Registry Dirctory Metrics should interface level

### PR DESCRIPTION
**Pull request by wxbty**
## What is the purpose of the change

interface level metrics of Dirctory num

## Verifying this change

image

### CODE REVIEW

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/AbstractDirectory.java [hunk 1]
```diff
@@ -386,7 +386,9 @@ private void refreshInvokers(BitList<Invoker<T>> targetInvokers, Collection<Invo
 
     @Override
     public void addDisabledInvoker(Invoker<T> invoker) {
-        MetricsEventBus.publish(DirectorSupport.disable(applicationModel));
+        if (invoker.getInterface() != null) {
+            MetricsEventBus.publish(DirectorSupport.disable(applicationModel, invoker.getInterface().getName()));
```
Comment (AlbumenJ) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/AbstractDirectory.java:390]: Can we replace with the exact invokers count in the collector after modify

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/AbstractDirectory.java (hunk 1 above)
Comment (wxbty) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/AbstractDirectory.java]: ok

**Comment by AlbumenJ** [comment]
Please fix conflicts

**Comment by wxbty** [comment]
> Please fix conflicts

done

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/AbstractDirectory.java [hunk 2]
```diff
@@ -499,4 +501,27 @@ protected String joinValidInvokerAddresses() {
...
+        return summaryMap;
+    }
+
+    private Map<String, Integer> groupByServiceKey(Collection<Invoker<T>> invokers) {
+
+        Map<String, Integer> serviceNumMap = new HashMap<>();
+        if (invokers.size() > 0) {
+            serviceNumMap = invokers.stream().filter(invoker -> invoker.getInterface() != null).collect(Collectors.groupingBy(invoker -> invoker.getInterface().getName(), Collectors.reducing(0, e -> 1, Integer::sum)));
+            if (invokers.stream().anyMatch(invoker -> invoker.getInterface() == null)) {
+                serviceNumMap.clear();
```
Comment (AlbumenJ) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/AbstractDirectory.java:521]: Why clear here

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/AbstractDirectory.java (hunk 2 above)
Comment (wxbty) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/AbstractDirectory.java]: image
An error will be reported without this line

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**